source/elements/oneTBB/source/algorithms/blocked_ranges/blocked_rangeNd_cls.rst

```diff
@@ -0,0 +1,175 @@
+.. SPDX-FileCopyrightText: 2019-2024 Intel Corporation
+.. SPDX-FileCopyrightText: Contributors to the oneAPI Specification project.
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+===============
+blocked_rangeNd
+===============
+**[algorithms.blocked_rangeNd]**
+
+Class template that represents a recursively divisible N-dimensional half-open interval.
+
+A ``blocked_rangeNd`` is the N-dimensional extension of ``blocked_range``, ``blocked_range2d``, and ``blocked_range3d``.
+You can interpret it as a Cartesian product of N instances of ``blocked_range``.
+Unlike ``blocked_range2d`` and ``blocked_range3d``, all ranges must be specified over the same ``Value`` type.
+Therefore, ``blocked_rangeNd<int,2>`` is a specialization of ``blocked_range2d`` where both dimensions are ``int``.
+
+.. code:: cpp
+
+    namespace oneapi {
+        namespace tbb {
+
+            template<typename Value, unsigned int N>
+            class blocked_rangeNd {
```

Comment:
should `blocked_rangeNd` be `blocked_range_nd`? I might understand why we named `blocked_range2d` and `3d` without underscore but for `_nd` we have a mix of `snake_case` and `camelCase` now

Reply:
Ah, the naming question! :)

To feed the discussion: SYCL has `nd_range` and `nd_item`, and C++23 has `mdspan` (but the real analogue to our class is called `extents` in C++).

Reply:
One more question to consider: should we also change "nd" to "md", following the pattern of C++ `mdspan`? This seems a bit more than purely cosmetic, as "md" stands for "multidimensional". And maybe even change the order of parts to `blocked_md_range`.

Reply:
I don't have a strong preference on a specific name. Here are my thoughts:

-  I definitely do NOT like `blocked_rangend`since that is indecipherable
-  and I agree that `blocked_rangeNd` is not great either because it appears to be camelCase unlike any other name. 

So I'm left with `blocked_range_nd`, `blocked_nd_range` and `blocked_md_range`.  Of these, I prefer the options with "nd" or "md" in the middle to differentiate even more from the existing blocked ranges. I have the slightest preference for "nd" over "md" simply because it doesn't imply multiple dimensions; it could be 1. "n" doesn't imply that it has to be "multi". But I wouldn't object to "md" if others have a preference for it.

Reply:
My preference is not `blocked_rangeNd` :)

`blocked_md_range`, `blocked_range_md`, `blocked_nd_range`, `blocked_range_nd` are all fine with me, with a different (but slight) level of preference.

I slightly prefer `md` to `nd`. Whether this `md` goes to an end to to a middle, I don't actually care. English-wise, perhaps `blocked_md_range` is better. Especially if we add aliases `blocked_2d_range` and `blocked_3d_range`. There is a caveat with those aliases, though... While the CTAD would work for `blocked_range_2d` and `_3d` since C++17, it would work for the respective aliases since C++20, which (IMO) looks weird from the user standpoint.

Reply:
> Hmm, I would prefer blocked_nd_range as it is a proper English word then. However we have blocked_range2d and blocked_range3d, so blocked_rangeNd is more consistent!

Looks like most if not all of us are OK with some inconsistency for the sake of a better name. We can think of `blocked_nd_range` as a new branch of `blocked_range` evolution, not a "descendant" of 2d/3d ranges (even though it borrows from those).

Reply:
I can life with that as long as this rationale is documented here explicitly, i.e. we explain it in the documentation.

Reply:
That's reasonable, but if you do not mind I would add it to the oneTBB Developer Guide instead of the specification. Perhaps here: https://oneapi-src.github.io/oneTBB/main/tbb_userguide/Advanced_Topic_Other_Kinds_of_Iteration_Spaces.html

Reply:
> I can life with that as long as this rationale is documented here explicitly, i.e. we explain it in the documentation.

Addressed by https://github.com/uxlfoundation/oneAPI-spec/pull/555/commits/ed26d0c785a828daf34a28c9249b079ba1c0856c. Also I have opened https://github.com/oneapi-src/oneTBB/issues/1529 issue for updating the oneTBB developer guide.